classes/class-wc-connect-service-schemas-store.php

```diff
@@ -263,19 +263,30 @@ public function get_predefined_packages_schema() {
 			return $predefined_packages;
 		}
 
+		/**
+		 * Returns the WooCommerce Shipping and WooCommerce Tax migration enabled status
+		 *
+		 * @return bool
+		 */
+		public function is_wcship_wctax_migration_enabled() {
+			$service_schemas = $this->get_service_schemas();
+
+			return is_object( $service_schemas ) && property_exists( $service_schemas, 'features' ) && property_exists( $service_schemas->features, 'wcshippingtax_upgrade_banner' ) && ! empty( $service_schemas->features->wcshippingtax_upgrade_banner );
+		}
+
 		/**
 		 * Returns the WooCommerce Shipping and WooCommerce Tax upgrade banners
 		 *
 		 * @return object|null The banners schema or null if no such id was found
 		 */
 		public function get_wcship_wctax_upgrade_banner() {
 			$service_schemas = $this->get_service_schemas();
-			// check if $service_schemas->features->wcshippingtax_upgrade_banner exists
-			if ( ! is_object( $service_schemas ) || ! property_exists( $service_schemas, 'features' ) || ! property_exists( $service_schemas->features, 'wcshippingtax_upgrade_banner' ) || empty( $service_schemas->features->wcshippingtax_upgrade_banner ) ) {
-				return null;
+
+			if ( $this->is_wcship_wctax_migration_enabled() ) {
+				return $service_schemas->features->wcshippingtax_upgrade_banner;
 			}
 
-			return $service_schemas->features->wcshippingtax_upgrade_banner;
+			return null;
```

Comment:
[Observation]

I have an observation because this part is updated, I don't think this would happen because our connect server will get updated before we push this plugin out. There is a slight chance `$this->get_service_schemas();` is cached and the feature isn't pulled from the server and this would return `null`.

When that happens, the $banner [here](https://github.com/Automattic/woocommerce-services/blob/wcs-migration/woocommerce-services.php#L1900) can be `null` and $banner->type, etc will throw undefined property error.

For that line, maybe we can do something defensive like
```
$banner = $schema->get_wcship_wctax_upgrade_banner();
if (empty($banner)) {
  return;
}
```